optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+
+import logging
+from typing import Any, Dict, Optional, Union
+
+from transformers import (
+    AutoConfig,
+    FeatureExtractionPipeline,
+    FillMaskPipeline,
+    Pipeline,
+    PreTrainedTokenizer,
+    PreTrainedTokenizerFast,
+    QuestionAnsweringPipeline,
+    SequenceFeatureExtractor,
+    TextClassificationPipeline,
+    TokenClassificationPipeline,
+)
+from transformers import pipeline as transformers_pipeline
+from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+from transformers.onnx.utils import get_preprocessor
+
+from optimum.neuron.modeling_base import NeuronBaseModel
+
+from ...modeling import (
+    NeuronModelForFeatureExtraction,
+    NeuronModelForMaskedLM,
+    NeuronModelForQuestionAnswering,
+    NeuronModelForSequenceClassification,
+    NeuronModelForTokenClassification,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+NEURONX_SUPPORTED_TASKS = {
+    "feature-extraction": {
+        "impl": FeatureExtractionPipeline,
+        "class": (NeuronModelForFeatureExtraction,),
+        "default": "distilbert-base-cased",
+        "type": "text",  # feature extraction is only supported for text at the moment
+    },
+    "fill-mask": {
+        "impl": FillMaskPipeline,
+        "class": (NeuronModelForMaskedLM,),
+        "default": "bert-base-cased",
+        "type": "text",
+    },
+    "question-answering": {
+        "impl": QuestionAnsweringPipeline,
+        "class": (NeuronModelForQuestionAnswering,),
+        "default": "distilbert-base-cased-distilled-squad",
+        "type": "text",
+    },
+    "text-classification": {
+        "impl": TextClassificationPipeline,
+        "class": (NeuronModelForSequenceClassification,),
+        "default": "distilbert-base-uncased-finetuned-sst-2-english",
+        "type": "text",
+    },
+    "token-classification": {
+        "impl": TokenClassificationPipeline,
+        "class": (NeuronModelForTokenClassification,),
+        "default": "dbmdz/bert-large-cased-finetuned-conll03-english",
+        "type": "text",
+    },
+}
+
+
+def load_pipeline(
+    model,
+    targeted_task,
+    load_tokenizer,
+    tokenizer,
+    feature_extractor,
+    load_feature_extractor,
+    SUPPORTED_TASKS,
+    input_shapes=None,
+    export=False,
+    subfolder: str = "",
+    use_auth_token: Optional[Union[bool, str]] = None,
+    revision: str = "main",
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    config: AutoConfig = None,
+    **kwargs,
+):
+    if model_kwargs is None:
+        model_kwargs = {}
+
+    # loads default model
+    if model is None:
+        model_id = SUPPORTED_TASKS[targeted_task]["default"]
+        if input_shapes is None:
+            input_shapes = {"batch_size": 1, "sequence_length": 128}
+            logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+        model = SUPPORTED_TASKS[targeted_task]["class"][0].from_pretrained(model_id, export=True, **input_shapes)
```

Comment:
`model_kwargs` could be compiler args, right? Just to confirm, so do we intend to set the default compiler args if the model is not given?

Reply:
Copied from original version of optimum. And even those are still model parameter since we use "ModelClass" would keep it for now.